### PR TITLE
ucai-core lib with base UC function client

### PR DESCRIPTION
**Pull request by serena-ruan**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**
Based on https://github.com/unitycatalog/unitycatalog/issues/525:
- Implement base UC function client

### CODE REVIEW

**Comment by serena-ruan** [comment]
@BenWilson2 Could you take a look at this PR?

**Comment by serena-ruan** [comment]
@dennyglee Hey Denny, could you take a look at this and see this folder make sense?

File: unitycatalog-ai/LICENSE.txt
Comment (dennyglee) [inline unitycatalog-ai/LICENSE.txt:1]: The root will have a `LICENSE.txt` so we should not include this license as well.

File: unitycatalog-ai/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Unity Catalog AI Core library
+
+The Unity Catalog AI Core library provides convenient APIs to interact with Unity Catalog functions, including functions creation, retrieval and execution.
```
Comment (BenWilson2) [inline unitycatalog-ai/README.md:3]:
```suggestion
The Unity Catalog AI Core library provides convenient APIs to interact with Unity Catalog functions, including the creation, retrieval and execution of functions.
```

File: unitycatalog-ai/src/ucai/core/client.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+_uc_function_client = None
+_client_lock = threading.Lock()
+
+
+@dataclass
+class FunctionExecutionResult:
+    """
+    Result of executing a function.
+    Value is always string, even if the function returns a scalar.
```
Comment (BenWilson2) [inline unitycatalog-ai/src/ucai/core/client.py:21]:
```suggestion
    Value is always string, even if the function returns a scalar or a collection.
```

File: unitycatalog-ai/README.md
Comment (dennyglee) [inline unitycatalog-ai/README.md:1]: Minor nit, could you clarify that ucai is part of the client interface for executing functions to support agents?

**Review by dennyglee — CHANGES_REQUESTED** [review]
Minor clarifications requested; the structure looks great!  Thanks @serena-ruan !

**Review by BenWilson2 — APPROVED** [review]
LGTM! Solid scaffolding :)

**Review by dennyglee — APPROVED** [review]

**Pull request merged**